impierce/ssi-agent
# feat: add support for OpenID4VP

# Description of change
See title

## Links to any relevant issues
#22 

## How the change has been tested
Added and improved aggregate tests.

How to manually test this change:
1. Simply run UniCore (`cargo run`)
2. Run UniMe on a physical device using this branch: https://github.com/impierce/identity-wallet/pull/197
3. Use the Postman collection to send both the `/v1/credentials` request and the `/v1/offers` request.
4. Render the encoded string that is returned as a QR code.
5. Scan the QR code with UniMe and accept the offer
6. Use the Postman collection to send the `v1/authorization_request` request.
7. Render the encoded string that is returned as a QR code.
8. Scan the QR code and accept the Authorization Request
9. UniCore will successfully verify UniMe's Authorization Response

## Definition of Done checklist
Add an `x` to the boxes that are relevant to your changes.
